spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb

```diff
@@ -251,6 +251,13 @@ class PodsProjectGenerator
                 installation_result.test_native_targets.count.should == 2
               end
 
+              it 'raises when a test spec has no source files' do
+                @watermelon_pod_target.test_spec_consumers.first.stubs(:source_files).returns([])
+                e = ->() { @installer.install! }.should.raise Informative
+                e.message.should.
+                  include 'Unable to install the `WatermelonLib` pod, because the `WatermelonLib-Unit-Tests` target in Xcode would have no sources to compile.'
+              end
```

Comment:
could you add a test for the non-test target case?

Reply:
sure!

Reply:
Actually this cannot be tested. The reason being is that `pod_target_installer.rb` checks for `unless target.should_build?` and therefore a target without any sources would be integrated via a `PBXAggregateTarget` instead.

Are you OK with that?

Reply:
Yeah makes sense. could / should `should_build?` be used here as well?

Reply:
I think its more of a `should_build_test_spec?` but I think its becoming convoluted. We really ought to split a separate pod target instance even for test specs maybe...`pod_target` is a god object.